pkg/pipelines/parameters_evaluator.go

```diff
@@ -43,46 +44,188 @@ func (e *parametersEvaluator) Run() error {
 func (e *parametersEvaluator) ExtractAll() {
 	e.ExtractPipelineName()
 	e.ExtractFromQueue()
-	e.ExtractFromGlobalSecrets()
 	e.ExtractFromSecrets()
+	e.ExtractFromBlockNames()
+	e.ExtractFromJobNames()
+	e.ExtractFromAgents()
+	e.ExtractFromJobMatrices()
+	e.ExtractFromParallelisms()
 }
 
 func (e *parametersEvaluator) ExtractPipelineName() {
 	e.tryExtractingFromPath([]string{"name"})
 }
 
+func (e *parametersEvaluator) ExtractFromQueue() {
+	e.tryExtractingFromPath([]string{"queue", "name"})
+
+	for index := range e.pipeline.QueueRules() {
+		e.tryExtractingFromPath([]string{"queue", strconv.Itoa(index), "name"})
+	}
+}
+
 func (e *parametersEvaluator) ExtractFromSecrets() {
+	e.ExtractFromGlobalSecrets()
+	e.ExtractFromBlockSecrets()
+	e.ExtractFromAfterPipelineSecrets()
+}
+
+func (e *parametersEvaluator) ExtractFromGlobalSecrets() {
+	e.extractFromSecretsAt(e.pipeline.GlobalJobConfig(), []string{"global_job_config"})
+}
+
+func (e *parametersEvaluator) ExtractFromBlockSecrets() {
 	for blockIndex, block := range e.pipeline.Blocks() {
-		secrets := block.Search("task", "secrets").Children()
-
-		for secretIndex := range secrets {
-			e.tryExtractingFromPath([]string{
-				"blocks",
-				strconv.Itoa(blockIndex),
-				"task",
-				"secrets",
-				strconv.Itoa(secretIndex),
-				"name",
-			})
+		e.extractFromSecretsAt(block.Search("task"), []string{"blocks", strconv.Itoa(blockIndex), "task"})
+	}
+}
+
+func (e *parametersEvaluator) ExtractFromContainerSecrets() {
+	for blockIndex := range e.pipeline.Blocks() {
+		e.tryExtractingFromPath([]string{"blocks", strconv.Itoa(blockIndex), "name"})
```

Comment:
The function mentions container secrets, but it is processing block names. One or the other seems to be wrong.